src/index.test.js

```diff
@@ -316,3 +316,50 @@ test('appends selected tags to aria-labelledby with text label', t => {
   t.deepEqual(wrapper.find('.dropdown-trigger').prop('aria-labelledby'), 'rdts_trigger rdts-0_tag')
   t.deepEqual(wrapper.find('.dropdown-trigger').prop('aria-label'), 'hello world')
 })
+
+test('select correct focused node when using external state data container', t => {
+  class DropdownContainer extends React.Component {
+    state = {
+      data: [
+        {
+          label: 'All data',
+          value: '0',
+        },
+        {
+          label: 'iWay',
+          value: '1',
+        },
+      ],
+    }
+    handleChange = ({ value, checked }) => {
+      this.setState(prevState => {
+        const { data } = prevState
+        data[value].checked = checked
+        return { data }
+      })
+    }
+    render() {
+      return <DropdownTreeSelect id={dropdownId} data={this.state.data} onChange={this.handleChange} />
+    }
+  }
+  const nodeAllData = {
+    _id: `${dropdownId}-0`,
+    _depth: 0,
+    label: 'All data',
+    value: '0',
+    children: undefined,
+    actions: [action],
+  }
+  const wrapper = mount(<DropdownContainer />)
+  const DropdownTreeSelectInstance = wrapper.find('DropdownTreeSelect')
+  DropdownTreeSelectInstance.instance().onCheckboxChange(nodeAllData._id, true)
+  t.deepEqual(DropdownTreeSelectInstance.state().currentFocus, nodeAllData._id)
+  t.deepEqual(DropdownTreeSelectInstance.state().tree.get(nodeAllData._id), {
```

Comment:
If we're checking for focused prop, then we should make that a specific check instead of comparing the whole state, no? I think that will convey intent better. Thoughts?

Reply:
Sure removing and fixing the test